addons/purchase/models/analytic_account.py

```diff
@@ -7,7 +7,7 @@
 class AccountAnalyticAccount(models.Model):
     _inherit = 'account.analytic.account'
 
-    purchase_order_count = fields.Integer("Purchase Order Count", compute='_compute_purchase_order_count')
+    purchase_order_count = fields.Integer("Purchase Order Count", compute='_compute_purchase_order_count', compute_sudo=True)
```

Comment:
I would use a group instead to hide the button if the user has no access right on it. Like here :
https://github.com/odoo/odoo/blob/9d5364c6bcac0df08549a8ce4e4e4b3b14d50e2f/addons/sale_mrp/models/mrp_production.py#L10-L13